UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump

```diff
@@ -98,17 +98,27 @@ public <<# if (uClass.getIsAbstract()) { append(realSb, "{0} ", "abstract"); } #
 for (StateMachine smq : uClass.getStateMachines())
   {
     if (smq.isQueued())
-    {
-      append(realSb," implements Runnable");
+    { 
+      if (uClass.hasParentInterface() == false){
```

Comment:
I just remember I have seen the case in which in the code generator we had two `implements` keywords which would make we cannot compile the java code. Now, I cannot get the example from the top of my head, but if now you're dealing with this case, you may want to check it as well.